test/run-tests

```diff
@@ -827,39 +850,80 @@ def setup_logging(config_logging):
 
     parser = argparse.ArgumentParser()
-    parser.add_argument("--secrets", default="/secrets", help="Directory with secrets")
     parser.add_argument(
-        "--config", default="/config", help="Directory with config.json"
+        "--secrets",
+        default=os.environ.get("TEST_HARNESS_SECRETS", "/secrets"),
+        help="Directory with secrets",
+    )
+    parser.add_argument(
+        "--config",
+        default=os.environ.get("TEST_HARNESS_CONFIG", "/config"),
+        help="Directory with config.json",
     )
     parser.add_argument(
-        "--cache", default="/cache", help="Directory for caching VM images"
+        "--cache",
+        default=os.environ.get("TEST_HARNESS_CACHE", "/cache"),
+        help="Directory for caching VM images",
     )
     parser.add_argument(
         "--inventory",
+        default=os.environ.get(
+            "TEST_HARNESS_INVENTORY",
+            "/usr/share/ansible/inventory/standard-inventory-qcow2",
+        ),
         help="Inventory to use for VMs",
-        default="/usr/share/ansible/inventory/standard-inventory-qcow2",
     )
+    default_pull_request = os.environ.get("TEST_HARNESS_PULL_REQUEST", None)
+    if default_pull_request:
+        default_pull_request = default_pull_request.split(",")
     parser.add_argument(
         "pull_request",
         nargs="*",
-        default=None,
+        default=default_pull_request,
         help="Pull requests to test. Example: " "linux-system-roles/network/1",
     )
     parser.add_argument(
         "--use-images",
-        default="*",
-        help="Test pull request only against images matching this pattern",
+        default=os.environ.get("TEST_HARNESS_USE_IMAGES", "*"),
+        help=(
+            "Test pull request only against images matching these patterns.  "
+            "This is a comma delimited list of fnmatch patterns which will be "
+            "applied to each image name and source.  If any of the patterns "
+            "match the name or source, the image will be included in testing."
+        ),
     )
     parser.add_argument(
         "--dry-run",
+        default=bool(strtobool(os.environ.get("TEST_HARNESS_DRY_RUN", "False"))),
+        action="store_true",
         help="Do not update pull request status or upload artifacts",
-        default=False,
+    )
+    parser.add_argument(
+        "--keep-results",
+        default=bool(strtobool(os.environ.get("TEST_HARNESS_KEEP_RESULTS", "False"))),
         action="store_true",
+        help="For debugging - do not remove the temp results directory",
+    )
+    parser.add_argument(
+        "--variant",
+        default=os.environ.get("TEST_HARNESS_VARIANT", ""),
+        help="Use this variant for PR test status e.g. staging",
+    )
+    parser.add_argument(
+        "--repositories",
+        default=os.environ.get("TEST_HARNESS_REPOSITORIES", None),
+        help="Comma delimited list of repositories to override config",
     )
 
     args = parser.parse_args()
```

Comment:
A few lines later there is a leftover from using "name" in the config previously, that should be removed.

Reply:
fixed in https://github.com/linux-system-roles/test-harness/pull/87/commits/709841e06909831fcf0a8ac493bce847e5242afa